### PR TITLE
Deepgram Flux turn detection for live voice, behind a default-off flag (JARVIS-1459)

### PR DESCRIPTION
**Pull request by alex-nork**
## Summary

Adds a `deepgram-flux` streaming STT provider speaking Deepgram's `/v2/listen` protocol, and lets live voice commit a turn on Flux's model-integrated `EndOfTurn` instead of the front-door `[0]` hold verdict, behind `liveVoice.flux.turnEnd.enabled` (default **false**).

Scope is turn detection only. The front door keeps `[1]` escalate and keeps ack/progress phrasing. Local VAD stays the barge-in trigger, the echo-adaptive gate stays upstream of Flux, and **no hold machinery is deleted**: it remains both the flag-off path and the fail-open fallback. Deleting it depends on the measurement this change exists to enable.

Assistant-only config. No LaunchDarkly registry entry, no platform terraform.

## What to do with it

`assistant/docs/flux-turn-detection-spike.md` is the runbook. Set `services.stt.provider: "deepgram-flux"` plus `liveVoice.flux.turnEnd.enabled: true`, using the same BYOK Deepgram key. A/B by flipping `turnEnd.enabled` with the provider held constant, and read `endpointCommitLatencyMs`, which is stamped from the same local-VAD speech-stop anchor on both paths.

**Read the runbook's warning before enabling.** Setting the provider to `deepgram-flux` disables batch transcription workspace-wide for the duration of the spike: voice messages, the `transcribe` and `media-processing` skills, inbound attachment transcription, and phone-call transcription. Set the provider back to `deepgram` to restore it.

## Corrections to the original analysis

The baseline in JARVIS-1459 was wrong and has been updated:

- Local VAD silence is **1200ms**, not 800ms. The 800 constant only applies to sessions built with no `liveVoice` config.
- On an **unheld** turn the endpoint-decider roundtrip is not additive, because the speculative front-door leg is already the assistant turn. The common-case bar Flux must beat is `silenceThresholdMs` alone.

## Measurement integrity

Four traps were found and closed, because the original metric would have produced a number that flattered Flux:

1. The two paths measured different spans. `endpointCommitLatencyMs` now records the same span on both.
2. The populations differed: `markEndpointDecision` fires from only two sites, so an unheld front-door turn emitted no event at all. The new field is stamped in `releaseUtterance`, which every committed turn passes through.
3. The speech-stop anchor did not exist on the front-door path, so a naive fix would have reported `0` on every arm-A turn.
4. A first utterance during a slow STT dial silently ran the old path while looking like a Flux session. The latch is now seeded from the configured provider before the dial.

## Notable bugs caught in review

- **A fast Flux turn-end was dropped as stale after any mid-thought pause**, stalling the turn for the full fail-open budget (~5.2s at shipped defaults) and then committing via the speculative hold path, mislabelled as `front-door`. Now disambiguated using Flux's own `turn_index`, with the VAD generation counter as fallback for unnumbered events.
- **Fatal `Error` frames were dropped**, so provider failures reached callers as a bare socket close. Now mapped to a categorized `error` event, with a latch so callers see exactly one error then `closed`.
- **`eager_eot_threshold` could exceed `eot_threshold`**, which Deepgram rejects at handshake.
- **A stale `turn-end` could commit while local VAD was routing resumed speech**, cutting off the caller's next words.

## Deliberately not done

- **`finalizeUtterance` is not implemented.** A no-op would report `finalized` without flushing, and since the flag defaults off the release always comes from the local VAD path, so every turn would dispatch with an empty transcript. Flux has no mid-stream flush; `CloseStream` per turn is required for correctness. The cost is a bounded teardown on perceived latency (not on `endpointCommitLatencyMs`, which is stamped before `stop()`).
- **Eager end-of-turn** stays off. Deepgram warns it raises LLM calls 50 to 70 percent, and the frames are parsed but unwired.
- **Managed/velay path, telephony, and languages beyond English** are all out of scope. `telephonyMode` is `"none"`.

## Known follow-ups

- The shared-transcriber `turn-end` branch routes to `pendingTranscriptCycle()`, which `handleFluxTurnEnd`'s first guard drops. Unreachable today since Flux never enters persistent mode, but it would ship untested if anyone revisits `finalizeUtterance`.
- `stt-stream-session.ts` has one more stale inline event enumeration around line 228.
- Per-turn latency has a known outlier in the unnumbered-fallback staleness path. Prefer medians over maxima.

## PRs merged into this branch

Plan PRs: #40189, #40187, #40188, #40196, #40197, #40206, #40213, #40220
Review fixes: #40224, #40225, #40226, #40227, #40229, #40232

Part of plan: flux-turn-detection.md

### CODE REVIEW

**Comment by alex-nork** [comment]
<details>
<summary>Plan papertrail: Deepgram Flux turn detection for live voice (JARVIS-1459)</summary>

# Deepgram Flux turn detection for live voice (JARVIS-1459)

## Overview

Add a `deepgram-flux` streaming STT provider that speaks Deepgram's `/v2/listen` protocol, and let live voice commit a turn on Flux's model-integrated `EndOfTurn` instead of the front-door `[0]` hold verdict — behind an assistant-only config flag, default off. Scope is turn detection only: the front door keeps `[1]` escalate and keeps ack/progress phrasing, local VAD keeps the barge-in trigger, and the echo-adaptive gate stays upstream of Flux. Nothing is deleted; the existing nova-2 + hold path remains the default and the fallback until Flux is measured.

**Not in scope (deliberate):** deleting the hold machinery, eager end-of-turn speculation, the managed/velay path, telephony, and language coverage beyond English. See "Deferred" at the end.

**Cross-repo:** none. `../vellum-assistant-platform` is present and was checked — this spike runs entirely on a BYOK Deepgram key through the daemon, so the velay relay (which pins the model server-side) and platform config are untouched. No platform plan file is needed.

## Resolved open questions

- **Language coverage.** Flux multi covers 10 languages; `clients/web/src/lib/stt/language-catalog.ts` is built around nova-3's monolingual roster and the daemon holds the authoritative list. The spike pins `flux-general-en` and does not touch the web catalog. Broad language support is deferred.
- **Chunk size.** Sample rate is negotiated per session via the start frame (`live-voice/protocol.ts:65`) and chunk cadence is client-driven, so Flux's 80ms recommendation is a client concern. The spike does not change capture; PR 6 logs observed chunk duration so PR 8's runbook can check it against 80ms before anyone tunes the client.
- **Eager end-of-turn.** Deferred. Deepgram warns it raises LLM calls 50–70%, and speculation is where the bugs live. PR 5's parser handles `EagerEndOfTurn` / `TurnResumed` so the follow-up is small, but `eagerEotThreshold` stays unset by default, which means Deepgram never emits them.

---

## PR 1: Extend the STT streaming contract with turn-detection events

### Depends on
None

### Branch
flux-turn-detect/pr-1-stt-turn-events

### Title
feat(assistant): add turn-detection events and deepgram-flux id to the STT contract

### Files
- `assistant/src/stt/types.ts`
- `assistant/src/live-voice/live-voice-session.ts`
- `assistant/src/calls/media-stream-stt-session.ts`
- `assistant/src/stt/__tests__/stt-types.test.ts` (new)

### Implementation steps
1. In `assistant/src/stt/types.ts`, add `"deepgram-flux"` to the `SttProviderId` union. Document inline that it shares the `deepgram` credential and is streaming-only.
2. In the same file, extend the `SttStreamServerEvent` union with four additive turn-detection variants, each documented as "emitted only by providers with model-integrated turn detection; providers without it never emit these":
   - `SttStreamServerTurnStartEvent` — `{ type: "turn-start" }`
   - `SttStreamServerEagerTurnEndEvent` — `{ type: "eager-turn-end"; text: string }`
   - `SttStreamServerTurnResumedEvent` — `{ type: "turn-resumed" }`
   - `SttStreamServerTurnEndEvent` — `{ type: "turn-end"; text: string; confidence?: number }`
3. Fix every consumer whose `switch` over `SttStreamServerEvent` now fails exhaustiveness. Known sites: `live-voice-session.ts:3118` and `live-voice-session.ts:3184`, plus the media-stream STT session. Add explicit no-op cases for the four new types with a comment that turn events are ignored until PR 7 wires them — **do not** collapse them into a `default:` branch, because the `never` check is what will force PR 7 to visit these sites.
4. Add `assistant/src/stt/__tests__/stt-types.test.ts` asserting the discriminated union accepts each new variant and that a consumer switch over all variants is exhaustive.

### Acceptance criteria
- `bun run typecheck` in `assistant/` passes with no exhaustiveness errors.
- Existing STT and live-voice tests pass unchanged — no runtime behavior changes.
- Feeding a `turn-end` event into the live-voice session's event handler is a no-op (no crash, no turn commit).

---

## PR 2: Add the `liveVoice.flux` config block

### Depends on
None

### Branch
flux-turn-detect/pr-2-flux-config-schema

### Title
feat(assistant): add liveVoice.flux config block for Flux turn detection

### Files
- `assistant/src/config/schemas/live-voice.ts`
- `assistant/src/config/__tests__/live-voice-schema.test.ts`

### Implementation steps
1. In `assistant/src/config/schemas/live-voice.ts`, add a `flux` block as a sibling of the existing `frontModel` block, following the same zod conventions (explicit `error` messages naming the full config path, `.describe()` on every field, `.default()` everywhere).
2. Fields:
   - `turnEnd.enabled` — boolean, **default `false`**. This is the feature flag. Describe it as: commit the live-voice turn on Flux's `EndOfTurn` instead of the front-door `[0]` hold verdict. Requires `services.stt.provider` to be `deepgram-flux`; ignored otherwise.
   - `model` — string, default `"flux-general-en"`.
   - `eotThreshold` — number, `.min(0.5).max(0.9)`, default `0.7`.
   - `eagerEotThreshold` — number, `.min(0.3).max(0.9)`, **optional with no default**. Describe: leaving it unset disables Deepgram's `EagerEndOfTurn` / `TurnResumed` events; enabling it raises LLM calls 50–70% from speculation.
   - `eotTimeoutMs` — integer, `.min(500).max(60000)`, default `5000`.
3. Wire the block into the parent live-voice schema with a `.default({})` so existing configs parse unchanged.
4. Add schema tests: defaults land as specified, out-of-range thresholds are rejected with the documented message, an empty config yields `turnEnd.enabled === false`, and `eagerEotThreshold` is absent (not zero) when unset.

### Acceptance criteria
- An existing workspace config with no `liveVoice.flux` key parses and yields `turnEnd.enabled === false`.
- `eotThreshold: 0.95` and `eotThreshold: 0.4` both fail validation with a message naming `liveVoice.flux.eotThreshold`.
- No migration is required (the block is fully defaulted).

---

## PR 3: Record which path made the endpoint decision

### Depends on
None

### Branch
flux-turn-detect/pr-3-endpoint-decision-source

### Title
feat(assistant): tag live-voice endpoint decisions with their source

### Files
- `assistant/src/live-voice/live-voice-metrics.ts`
- `assistant/src/live-voice/__tests__/live-voice-metrics.test.ts`

### Implementation steps
1. In `assistant/src/live-voice/live-voice-metrics.ts`, add `export type VoiceEndpointSource = "front-door" | "flux";`.
2. Add an optional `source?: VoiceEndpointSource` to the `LiveVoiceEndpointDecisionMark` interface (near line 33). Default it to `"front-door"` inside `markEndpointDecision` so existing callers are unchanged.
3. Track the source on the per-turn metrics record and surface it on `LiveVoiceMetricsFrame` as `endpointDecisionSource?: VoiceEndpointSource`, alongside the existing `endpointHoldCount` / `endpointDecisionMaxLatencyMs` fields (lines ~120–124, ~170–171, ~190–193). Follow the existing convention where these fields are absent unless the decider was consulted.
4. Register the new field in `getLiveVoiceMetricsAggregateFields` (line ~542) if it belongs there — check how the existing endpoint fields are handled and match.
5. Tests: a decision with no `source` reports `"front-door"`; a `"flux"` decision reports `"flux"`; latency and hold-count accounting are unchanged.

### Acceptance criteria
- `markEndpointDecision(turnId, { action: "hold", latencyMs: 900 })` still works with no source and reports `"front-door"`.
- The emitted metrics frame carries `endpointDecisionSource` only when an endpoint decision actually happened.
- Existing live-voice metrics tests pass unchanged.

---

## PR 4: Register `deepgram-flux` in the provider catalog and config

### Depends on
PR 1

### Branch
flux-turn-detect/pr-4-flux-provider-registration

### Title
feat(assistant): register the deepgram-flux STT provider

### Files
- `assistant/src/providers/speech-to-text/provider-catalog.ts`
- `assistant/src/config/schemas/stt.ts`
- `assistant/src/stt/daemon-batch-transcriber.ts`
- `assistant/src/providers/speech-to-text/__tests__/provider-catalog.test.ts`

### Implementation steps
1. Follow `assistant/docs/stt-provider-onboarding.md` steps 1, 3, 4, and 6. Add a `CATALOG` entry in `provider-catalog.ts` keyed `"deepgram-flux"`:
   - `displayName: "Deepgram Flux"`, `subtitle` naming conversational turn detection, `setupMode: "api-key"`.
   - `credentialProvider: "deepgram"` — **shared key with the existing `deepgram` provider**. No new credential, and `listCredentialProviderNames()` dedupes it automatically (onboarding step 5 needs no change).
   - `supportedBoundaries: new Set(["daemon-streaming"])` — streaming only; Flux has no batch endpoint.
   - `telephonyMode: "none"` for the spike. Do not route phone calls through Flux yet.
   - `conversationStreamingMode: "realtime-ws"`, `supportsDiarization: false`, `languageSelection: "manual"`.
   - `credentialsGuide` pointing at `https://console.deepgram.com/`, matching the existing `deepgram` entry.
2. In `assistant/src/config/schemas/stt.ts`, append `"deepgram-flux"` to the `VALID_STT_PROVIDERS` tuple (line 7). No alias entry — the ID is explicit.
3. In `assistant/src/stt/daemon-batch-transcriber.ts`, add a `case "deepgram-flux"` to the exhaustive switch in `createDaemonBatchTranscriber()` that throws a clear "Flux is streaming-only; use the deepgram provider for batch transcription" error. The `never` check will not compile without it.
4. Update `__tests__/provider-catalog.test.ts` for the new entry: it appears in `listProviderIds()`, `getCredentialProvider("deepgram-flux") === "deepgram"`, `supportsBoundary("deepgram-flux", "daemon-batch") === false`, and `listCredentialProviderNames()` contains exactly one `"deepgram"`.

### Acceptance criteria
- Setting `services.stt.provider: "deepgram-flux"` passes config validation.
- The provider appears in `GET /v1/stt/providers` with the shared-key behavior (clients derive shared-vs-exclusive from `apiKeyProviderName != id`).
- Requesting batch transcription on `deepgram-flux` fails with the explicit streaming-only message, not a `never`-check crash.
- Any provider-parity tests referenced by onboarding step 7 pass.

---

## PR 5: Flux frame parser

### Depends on
PR 1

### Branch
flux-turn-detect/pr-5-flux-frame-parser

### Title
feat(assistant): parse Deepgram Flux v2 frames into STT stream events

### Files
- `assistant/src/providers/speech-to-text/deepgram-flux-frames.ts` (new)
- `assistant/src/providers/speech-to-text/__tests__/deepgram-flux-frames.test.ts` (new)

### Implementation steps
1. Create `deepgram-flux-frames.ts` holding **only pure functions** — no socket, no I/O. This is the risky protocol surface, isolated so it can be tested exhaustively.
2. Define the wire types for Flux's `/v2/listen` frames: `Connected`, `StartOfTurn`, `Update`, `TurnInfo`, `EagerEndOfTurn`, `TurnResumed`, `EndOfTurn`. `TurnInfo` carries `transcript` and a `words` array with per-word timings and confidence.
3. Export `parseFluxFrame(raw: unknown): SttStreamServerEvent[]` mapping each frame to zero or more contract events:
   - `Connected` → `[]`
   - `StartOfTurn` → `[{ type: "turn-start" }]`
   - `Update` → `[{ type: "partial", text }]`
   - `TurnInfo` → `[{ type: "partial", text }]` (interim transcript refresh; the commit comes from `EndOfTurn`)
   - `EagerEndOfTurn` → `[{ type: "eager-turn-end", text }]`
   - `TurnResumed` → `[{ type: "turn-resumed" }]`
   - `EndOfTurn` → `[{ type: "final", text }, { type: "turn-end", text, confidence }]` — emit the `final` **first** so consumers that ignore turn events still commit the transcript exactly as they do today.
4. Unknown or malformed frames return `[]` and log at debug — never throw. Deepgram can add frame types without breaking us.
5. Export `buildFluxQueryParams(opts)` producing the `/v2/listen` query string from `{ model, encoding, sampleRate, eotThreshold, eagerEotThreshold, eotTimeoutMs, languageHint }`. Omit `eager_eot_threshold` entirely when undefined — that is what keeps eager events off.
6. Tests: a fixture frame per type asserting the exact event sequence; `EndOfTurn` yields `final` before `turn-end`; malformed input yields `[]`; `buildFluxQueryParams` omits `eager_eot_threshold` when unset and clamps/passes thresholds correctly.

### Acceptance criteria
- Every documented Flux frame type has a passing fixture test.
- `parseFluxFrame` never throws on arbitrary input (cover with a garbage-input test).
- The module imports nothing from the socket layer — verifiable by inspection of its imports.

---

## PR 6: Flux streaming transcriber

### Depends on
PR 4, PR 5

### Branch
flux-turn-detect/pr-6-flux-streaming-transcriber

### Title
feat(assistant): add the Deepgram Flux streaming transcriber

### Files
- `assistant/src/providers/speech-to-text/deepgram-flux-realtime.ts` (new)
- `assistant/src/providers/speech-to-text/resolve.ts`
- `assistant/src/providers/speech-to-text/__tests__/deepgram-flux-realtime.test.ts` (new)

### Implementation steps
1. Create `deepgram-flux-realtime.ts` implementing `StreamingTranscriber` with `providerId: "deepgram-flux"` and `boundaryId: "daemon-streaming"`. Model the socket lifecycle, keepalive, and reconnect on the existing `deepgram-realtime.ts` — but dial `wss://api.deepgram.com/v2/listen` and delegate all frame handling to `parseFluxFrame` from PR 5.
2. Read tuning from `liveVoice.flux` (PR 2) at construction: `model`, `eotThreshold`, `eagerEotThreshold`, `eotTimeoutMs`. Build the URL with `buildFluxQueryParams`.
3. `finalizeUtterance` — Flux owns turn boundaries, so do **not** implement it. Leaving the optional method off makes callers feature-detect and fall back to `stop()`, which is the correct semantics.
4. Log the observed audio chunk duration once per session at debug (derived from byte length and the negotiated sample rate) so PR 8's runbook can compare it against Flux's 80ms recommendation. Do not change capture.
5. Map socket and auth failures onto `SttError` categories exactly as `deepgram-realtime.ts` does (`auth`, `rate-limit`, `timeout`, `provider-error`), and emit `closed` on teardown.
6. In `resolve.ts`, add `deepgram-flux` to `resolveStreamingTranscriber` so it constructs this transcriber. Keep the existing boundary guard: the catalog's `telephonyMode: "none"` (PR 4) already makes boundary-requiring telephony callers resolve to `null` and fall back — verify that with a test rather than adding a new conditional.
7. Tests against a mock socket: session opens with the expected URL and params; audio chunks are forwarded; a scripted frame sequence produces the expected `SttStreamServerEvent` stream; auth failure surfaces `SttError` with category `auth`; telephony resolution returns `null`.

### Acceptance criteria
- `resolveStreamingTranscriber({ providerId: "deepgram-flux" })` returns a working transcriber when a Deepgram key is present, and `null` when it is absent.
- The dialed URL is `/v2/listen` and contains `model=flux-general-en`, `eot_threshold`, `eot_timeout_ms`, and **no** `eager_eot_threshold` under default config.
- A scripted `EndOfTurn` frame produces a `final` followed by a `turn-end`.
- Telephony callers resolve to `null` (Flux is not wired into phone calls in this spike).

---

## PR 7: Commit live-voice turns on Flux's EndOfTurn

### Depends on
PR 2, PR 3, PR 6

### Branch
flux-turn-detect/pr-7-flux-turn-end-integration

### Title
feat(assistant): commit live-voice turns on Flux end-of-turn behind a flag

### Files
- `assistant/src/live-voice/live-voice-session.ts`
- `assistant/src/live-voice/__tests__/live-voice-flux-turn-end.test.ts` (new)

### Implementation steps
1. In `live-voice-session.ts`, compute a per-session `fluxTurnEndActive` latch at session start: `liveVoice.flux.turnEnd.enabled === true` **and** the resolved streaming provider is `deepgram-flux`. Anything else leaves every existing code path byte-for-byte unchanged.
2. Replace the PR 1 no-op cases in the two event switches (`:3118`, `:3184`) with real handling, gated on the latch:
   - `turn-start` — no-op for now; local VAD still owns barge-in. Leave a comment saying so explicitly.
   - `turn-end` — commit the utterance immediately: run the same commit path the current silence boundary takes after a `release`, and dispatch the front-door leg with `speculativeHoldAllowed: false` so `frontDoorDecisionRule` is built with `includeHold: false` and the model is told the caller has finished. The front door still returns `[1]` escalate or a committed answer — only the `[0]` branch is bypassed.
   - `eager-turn-end` / `turn-resumed` — remain no-ops. Add a comment pointing at the deferred eager follow-up.
3. When the latch is active, do **not** start the silence-boundary hold path: skip the endpoint-decision LLM call and the `endpointExtensionMs` / `endpointMaxExtensions` extension-and-replay machinery. Leave all of that code in place, untouched, for the flag-off path.
4. Keep the local VAD barge-in trigger active in both modes (`DEFAULT_SILENCE_THRESHOLD_MS` at `:137` and the sustained-speech run logic at `:134-147`). Add a comment recording why: local energy detection beats a network roundtrip for interrupt-during-TTS, and the echo-adaptive gate must stay upstream of Flux because Flux cannot hear our own TTS and would read echoed audio as a user turn.
5. Fail open. If the latch is active but no `turn-end` arrives within `eotTimeoutMs + a margin` after speech stops, fall back to the existing silence-boundary path for that utterance and log a warning. A Flux outage must degrade to today's behavior, not to a hung turn.
6. Record the decision through `markEndpointDecision(turnId, { action: "release", latencyMs, source: "flux" })` using PR 3's field, where `latencyMs` is measured from the local VAD's speech-stop mark to the `turn-end` event. That is what makes the two paths comparable in one metrics frame.
7. Tests in a new `live-voice-flux-turn-end.test.ts`: flag off → turn events ignored and the hold path runs (assert the existing behavior is untouched); flag on + `turn-end` → turn commits without an endpoint-decision LLM call, and the front-door leg is built with `includeHold: false`; flag on + no `turn-end` → falls back to the silence path after the timeout; flag on → `endpointDecisionSource` is `"flux"`; barge-in during TTS still fires from local VAD in both modes.

### Acceptance criteria
- With `liveVoice.flux.turnEnd.enabled: false` (the default), live-voice behavior is identical to today — verified by the full existing live-voice suite passing unchanged.
- With the flag on and Flux resolved, a completed utterance commits on `turn-end` with no endpoint-decision LLM call in the turn.
- The `[1]` escalate verdict and ack/progress phrasing still work with the flag on.
- Barge-in during TTS fires from local VAD in both modes.
- A Flux stream that never emits `turn-end` degrades to the existing silence-boundary path rather than hanging.

---

## PR 8: Local spike runbook

### Depends on
PR 7

### Branch
flux-turn-detect/pr-8-flux-spike-runbook

### Title
docs(assistant): add the Flux turn-detection spike runbook

### Files
- `assistant/docs/flux-turn-detection-spike.md` (new)

### Implementation steps
1. Write the runbook for measuring Flux against the current path locally:
   - Config to enable: `services.stt.provider: "deepgram-flux"` plus `liveVoice.flux.turnEnd.enabled: true`, and where the BYOK Deepgram key goes (same credential as the existing `deepgram` provider — no new key needed).
   - How to A/B: flip `turnEnd.enabled` between runs with the provider left on `deepgram-flux`, so the STT engine is held constant and only the turn-end path changes. Note that flipping the provider instead confounds the STT model with the turn-end path.
   - What to read: the `endpoint_decision` metrics event, with `endpointDecisionSource` distinguishing the paths and `endpointDecisionMaxLatencyMs` as the headline number. Include the baseline to beat — 800ms local VAD silence plus the ~670–1130ms endpoint-decider roundtrip documented at `config/schemas/live-voice.ts:161`, against Flux's claimed sub-400ms.
   - Where to find the observed chunk-duration log line from PR 6, and the note that Flux recommends 80ms.
2. Document the known gaps so the next person does not rediscover them: English only, no telephony, no managed/velay path, eager EOT off, hold machinery still present and still the fallback.
3. Add a short "what would justify deleting the hold path" section — the measured bar Flux has to clear before the follow-up cleanup is worth doing.

### Acceptance criteria
- Someone with a Deepgram key can enable Flux, run a live-voice session, and read a latency comparison without asking questions.
- The runbook states plainly that the hold machinery is untouched and how to fall back.

---

## Deferred (explicitly out of scope)

- **Deleting the `[0]` hold path** — `HOLD_VERDICT_TOKEN`, the `includeHold` branch of `frontDoorDecisionRule`, the `holdEnabled` param on `classifyFrontDoorLeading`, the speculative-dispatch state in `live-voice-session.ts` (`speculativePending`, `speculativeGeneration`, `speculativeContent`, `speculativeDispatchedAtMs`, `speculativeHoldAllowed`, `speculativeBuffer`), and `frontModel.endpointExtensionMs` / `endpointMaxExtensions`. Only after Flux is measured.
- **Eager end-of-turn** — `eagerEotThreshold` plus wiring `eager-turn-end` / `turn-resumed` onto the existing speculative dispatch + rollback machinery, which is already shaped for it.
- **Managed / velay path** — the relay pins the model server-side, so managed rollout is a relay change and can A/B without a client release.
- **Telephony** — `telephonyMode` stays `"none"`.
- **Language coverage** — `flux-general-multi` and the web language catalog.

</details>

File: assistant/src/config/bundled-skills/settings/TOOLS.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
...he web voice picker). For deepgram, pass a Deepgram Aura model ID. Use fish_audio_reference_id for Fish Audio voice reference. Use stt_language to set the spoken language for speech recognition: one of the 50 base language codes on the verified Deepgram nova-3 monolingual roster (e.g. en, es, hi, ta, zh, ko), or \"multi\" for code-switching mid-sentence across its 10-language roster (English, Spanish, French, German, Hindi, Russian, Portuguese, Japanese, Italian, Dutch; e.g. Hinglish); plain language names like \"tamil\" or \"multilingual\" are accepted and normalized. Accepted values follow the configured STT provider: vellum-managed and deepgram accept the full roster plus \"multi\"; xai accepts only the 10 multilingual-roster codes (en, es, fr, de, hi, ru, pt, ja, it, nl) and rejects \"multi\" and the extended codes (both are verified for Deepgram nova-3 only); google-gemini / openai-whisper auto-detect natively and deepgram-flux runs an English-only model, so the value persists but is ignored while they are active. Deepgram uses the same API key for TTS and STT, and deepgram-flux shares it too. Changes persist to services.stt / services.tts config and take effect immediately.",
```
[the remark this answers is not in the document]
Comment (alex-nork) [reply assistant/src/config/bundled-skills/settings/TOOLS.json]: Addressed in direct push to alex-nork/flux-turn-detection
